### PR TITLE
feat: show status of community members

### LINKED ISSUE #2811: Agree on Spec & implement user status on a community
To support seeing who is online or not in a community it should be possible to do this with a under the hood channel that each client can send a ping every X minutes, when the app quits it can send a offline ping, no ping in the last X minutes can interpreted as a offline state. We should talk to @cammellos to agree on whatever method is chosen before implementing.

### PR DESCRIPTION
**Pull request by richard-ramos**
Fixes #2811

### CODE REVIEW

**Comment by anastasiyaig** [comment]
Screenshot 2021-07-12 at 15 40 31


for some reason it shows different states when both instances are up and running and sent some messages in the chat. Also, i believe it was never designed so not sure about the UI.

File: src/app/chat/signal_handling.nim [hunk 1]
```diff
@@ -4,7 +4,7 @@ import
 proc handleSignals(self: ChatController) =
   self.status.events.on(SignalType.Message.event) do(e:Args):
     var data = MessageSignal(e)
-    self.status.chat.update(data.chats, data.messages, data.emojiReactions, data.communities, data.membershipRequests, data.pinnedMessages, data.activityCenterNotification)
+    self.status.chat.update(data.chats, data.messages, data.emojiReactions, data.communities, data.membershipRequests, data.pinnedMessages, data.activityCenterNotification, data.statusUpdates)
```
Comment (jrainville) [inline src/app/chat/signal_handling.nim:7]: https://media.giphy.com/media/1jXGsHY2EKdL27mEMd/giphy.gif

File: ui/app/AppLayouts/Profile/Sections/AdvancedContainer.qml [hunk 2]
```diff
@@ -132,6 +133,21 @@ Item {
...
+        Loader {
+            active: onlineUsers.switchChecked
+            anchors.left: parent.left
+            anchors.right: parent.right
+            sourceComponent: StatusSettingsLineButton {
+                text: qsTr("Broadcast user status")
+                isSwitch: true
+                switchChecked: profileModel.profile.sendUserStatus
+                onClicked: function (checked) {
+                    profileModel.setSendUserStatus(checked)
```
Comment (jrainville) [inline ui/app/AppLayouts/Profile/Sections/AdvancedContainer.qml:145]: Might be worth to add a check here to not call the function uselessly:
```qml
if (profileModel.profile.sendUserStatus !== checked) {
  profileModel.setSendUserStatus(checked)
}
```

**Review by jrainville — APPROVED** [review]
Nice work!

File: src/app/chat/event_handling.nim [hunk 3]
```diff
@@ -9,7 +9,12 @@ import # status-desktop libs
 proc handleChatEvents(self: ChatController) =
   # Display already saved messages
   self.status.events.on("messagesLoaded") do(e:Args):
-    self.view.pushMessages(MsgsLoadedArgs(e).messages)
+    let evArgs = MsgsLoadedArgs(e)
+    self.view.pushMessages(evArgs.messages)
+    for statusUpdate in evArgs.statusUpdates:
+      echo "updating member visibility", $statusUpdate
```
Comment (jrainville) [inline src/app/chat/event_handling.nim:15]: ✂️ 
```suggestion
```

**Review by jrainville — APPROVED** [review]

**Comment by iurimatias** [comment]
this should not be merged until the corresponding PR in status-go is also merged https://github.com/status-im/status-go/pull/2276

File: src/status/chat/chat.nim [hunk 4]
```diff
@@ -107,6 +107,17 @@ type CommunityCategory* = object
...
 
+type StatusUpdateType* {.pure.}= enum
+  Unknown = 0,
+  Online = 1, 
+  DoNotDisturb = 2
+
+type StatusUpdate* = object
+  publicKey*: string
+  statusType*: StatusUpdateType
+  clock*: int64
```
Comment (pinebit) [inline src/status/chat/chat.nim:118]: this is `uint64` in protobuf (status-go)

File: ui/app/AppLayouts/Chat/ChatColumn/User.qml [hunk 5]
```diff
@@ -80,16 +81,16 @@ Item {
             width: 10
             radius: 20
             color: {
-                let lastSeenMinutesAgo = (currentTime - parseInt(lastSeen)) / 1000 / 60
+                let lastSeenMinutesAgo = (currentTime/1000 - parseInt(lastSeen)) / 60
```
Comment (pinebit) [inline ui/app/AppLayouts/Chat/ChatColumn/User.qml:84]: `parseUInt`?

**Pull request merged**